ethsigner/app/src/main/java/tech/pegasys/ethsigner/FileBasedTransactionSignerCommand.java

```diff
@@ -50,28 +55,23 @@ public FileBasedSignerCliConfig() {}
       description = "The path to a file containing the key used to sign transactions.",
       required = true,
       arity = "1")
-  private Path keyFile;
-
-  @Override
-  public Path getPasswordFilePath() {
-    return passwordFilePath;
-  }
-
-  @Override
-  public Path getKeyPath() {
-    return keyFile;
-  }
-
-  @Override
-  public boolean isConfigured() {
-    return passwordFilePath != null && keyFile != null;
-  }
+  private Path keyFilePath;
 
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("passwordFilePath", passwordFilePath)
-        .add("keyFile", keyFile)
+        .add("keyFilePath", keyFilePath)
         .toString();
   }
+
+  @Override
+  public void run() {
```

Comment:
Be nice if somehow this function was only responsible for building the signer required, and the base class creates the EthSigner based off of the ParentCommand/Config.

This means the code to create the EthSigner would be reused between FileBased and Hashicorp.

This common code would also probably run the ethsigner....

Reply:
I've changed that. The commandLineConfig is starting EthSigner now, the TransactionSigner Commands just pass in the transaction signer.